src/server/game/Handlers/MiscHandler.cpp

```diff
@@ -1144,3 +1144,30 @@ void WorldSession::HandleCloseInteraction(WorldPackets::Misc::CloseInteraction&
     if (_player->PlayerTalkClass->GetInteractionData().SourceGuid == closeInteraction.SourceGuid)
         _player->PlayerTalkClass->GetInteractionData().Reset();
 }
+
+void WorldSession::HandleAdventureJournalOpenQuest(WorldPackets::Misc::AdventureJournalOpenQuest& packet)
+{
+    auto adventureJournalEntry = sAdventureJournalStore[packet.AdventureJournalID];
+    if (!adventureJournalEntry)
+        return;
+
+    Quest const* quest = sObjectMgr->GetQuestTemplate(adventureJournalEntry->QuestID);
+    if (!quest)
+        return;
+
+    if (_player->hasQuest(adventureJournalEntry->QuestID) && _player->CanTakeQuest(quest, true))
+    {
+        PlayerMenu menu(_player->GetSession());
+        menu.SendQuestGiverQuestDetails(quest, _player->GetGUID(), true, false);
+    }
+}
+
+void WorldSession::HandleAdventureJournalStartQuest(WorldPackets::Misc::AdventureJournalStartQuest& packet)
+{
+    Quest const* quest = sObjectMgr->GetQuestTemplate(packet.QuestID)
+    if (!quest)
+        return;
+
+    if (!_player->hasQuest(packet.QuestID))
+        _player->AddQuestAndCheckCompletion(quest, nullptr);
```

Comment:
this did not fix any exploit. You either have to check if the questid is in adventurejournal db2 or maybe theres even a fitting flag for it